lib/media/streaming_engine.js

```diff
@@ -1953,6 +1959,29 @@ shaka.media.StreamingEngine = class {
     shaka.log.v2(logPrefix, 'appended media segment');
   }
 
+  /**
+   * @param {shaka.media.StreamingEngine.MediaState_} mediaState
+   * @param {shaka.media.SegmentReference|shaka.media.InitSegmentReference}
+   *   reference
+   * @return {!Promise}
+   * @private
+   */
+  async changeTypeIfNeeded_(mediaState, reference) {
+    // TODO: Do I also need to handle new init segments for this content, or
+    // do we handle that already? The test content I have is all
+    // self-initializing segments.
+    if (reference.codecs) {
```

Comment:
Since there is a codec/container change, it is always necessary to give a new initialization segment.

Reply:
Actually, on further inspection, it looks like `initSourceBuffer_` is called before each segment is appended, and that should handle the init segment changing. So I don't need to do anything else.